File: codewars/6kyu/binary-to-text-(ascii)-conversion/dduck-binary-to-text-(ascii)-conversion.js
// https://www.codewars.com/kata/5583d268479559400d000064/

const binaryToString = (binary, result = []) => {
  if (binary.length === 0) {
    return result.join('');
  }
  result.push(String.fromCharCode(parseInt(binary.slice(0, 8), 2)));
  return binaryToString(binary.slice(8), result);
};


// Test apha chars
console.log(binaryToString('01100001'), 'a');
console.log(binaryToString('01001011010101000100100001011000010000100101100101000101'), 'KTHXBYE');

// Test numeric
console.log(binaryToString('00110001001100000011000100110001'), '1011');

// Test special chars
console.log(binaryToString('001111000011101000101001'), '<:)');

File: codewars/6kyu/binary-to-text-(ascii)-conversion/dduck-binary-to-text-(ascii)-conversion.py
# https://www.codewars.com/kata/5583d268479559400d000064/

from unittest import TestCase

binary_to_string = lambda b, bit=8: "".join([chr(int(b[i:i + bit], 2)) for i in range(0, len(b), bit)])

TestCase().assertEqual(binary_to_string('0100100001100101011011000110110001101111'), 'Hello')
TestCase().assertEqual(binary_to_string('00110001001100000011000100110001'), '1011')
